### PR TITLE
refactor/1697 introduce parse function which returns ParsedCommand

### LINKED ISSUE #1697: Introduce parse function which returns ParsedCommand
This has two advantages:

1. By renaming, the call site in `InputController.js` gets more readable (see the PR)
2. This keeps the diff low as

in  a following PR, I want to split  `ParsedCommand` into two parts (one which holds the data used in the `InputController` and one Parser class which does the parsing/validating). By this, we will have a clear processing pipeline of reading from the CLI -> parsing -> Error or Command returned.

--- 
forget it
I just noticed I can do it in one PR by introducing a `parse` function which returns a `parsedCommand`

### PR DESCRIPTION
**Pull request by JonasCir**
Closes #1697 
By this, the call site in `InputController.js` gets more readable. 
Consider this PR as a trick to reduce the diff of the following PR. Once this is merged, I can start to separate the parsing process from the command that is returned and handled.

### CODE REVIEW

**Comment by JonasCir** [comment]
I just had an idea...

**Review by n8rzz — COMMENTED** [review]
I’m not sure I understand the name change here. `CommandParser` describes and action. It does something that the name describes. `ParsedCommand`, however, suggests a thing has already been done. This name describes an immutable thing. 

Perhaps that’s the direction?  I want to call it out now, tho, cause naming is incredibly important.

**Comment by JonasCir** [comment]
@n8rzz Your totally right, I updated the PR now such that there is an actual `parse` method in the parser which returns the `ParsedCommand`. This should be more intuitive and provides us with a clean interface.

**Comment by JonasCir** [comment]
This is better :) The parser logic is now separated from the command that is returned.

File: test/commands/ParsedComman.spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import _isEqual from 'lodash/isEqual';
```
Comment (n8rzz) [inline test/commands/ParsedComman.spec.js:1]: missing a `d` in the filename 😉

Comment (JonasCir) [reply test/commands/ParsedComman.spec.js]: good eye :)

File: src/assets/scripts/client/InputController.js [hunk 2]
```diff
@@ -778,29 +778,29 @@ export default class InputController {
...
+            cmd = parser.parse();
         } catch (error) {
             UiController.ui_log('Command not understood', true);
-
             throw error;
         }
-
-        if (aircraftCommandParser.command !== PARSED_COMMAND_NAME.TRANSMIT) {
-            return this.processSystemCommand(aircraftCommandParser);
+        const parsedCommand = cmd;
```
Comment (n8rzz) [inline src/assets/scripts/client/InputController.js:795]: Why are we reassigning here?  Could we continue to use `cmd`?

File: test/commands/ParsedCommand.spec.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import _isEqual from 'lodash/isEqual';
+import ava from 'ava';
+import CommandParser from '../../src/assets/scripts/client/commands/parsers/CommandParser';
+
+
```
Comment (n8rzz) [inline test/commands/ParsedCommand.spec.js:5]: Please use a single line for spacing. Here and below too

**Review by n8rzz — CHANGES_REQUESTED** [review]
Just a couple small things and you’re good to go.

File: src/assets/scripts/client/InputController.js (hunk 2 above)
Comment (JonasCir) [reply src/assets/scripts/client/InputController.js:795]: the reason is that I want the parsed command to be `const`, but I cannot do it in the try block as it is then only scoped the block. To make it `const`, I need to trick and use a reassign here.

File: test/commands/ParsedCommand.spec.js (hunk 3 above)
Comment (JonasCir) [reply test/commands/ParsedCommand.spec.js]: Done :)

**Review by n8rzz — APPROVED** [review]

**Pull request merged**